src/blocks/recommendations/class-recommendations-block.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Parse.ly Recommendations Block class file.
+ *
+ * @package Parsely
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely;
+
+/**
+ * Parse.ly Recommendations Block for the WordPress Block Editor.
+ *
+ * @since 3.2.0
+ */
+class Recommendations_Block {
+	const MINIMUM_WORDPRESS_VERSION = '5.6';
+
+	/**
+	 * Determine whether the block and its assets should be registered.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		global $wp_version;
+
+		if ( ! apply_filters( 'wp_parsely_recommendations_block_enabled', false ) ) {
+			// This block is behind a "feature flag" and it's not enabled. Bail.
+			return;
+		}
+
+		if ( ! isset( $wp_version ) || version_compare( $wp_version, self::MINIMUM_WORDPRESS_VERSION ) < 0 ) {
+			// WordPress is not recent enough to run this block.
+			return;
+		}
+
+		// Temporary workaround - remove block when in FSE due to issues.
+		global $pagenow;
+		if ( 'site-editor.php' === $pagenow ) {
+			add_action(
+				'admin_init',
+				function() {
+					unregister_block_type( 'wp-parsely/recommendations' );
+				}
+			);
+		}
```

Comment:
I wonder if we could do this check inside `register_block_and_assets` and instead of regitering and then unregistering, we could not register at all in the first place.

Reply:
I think I had the same idea, but I don't remember why I didn't do it. Will recheck this.

Reply:
I guess the reason I did it was for it to be grouped with the other code that would not register the Block (no filter, old WP version). But what you actually suggest is cleaner, and I'm pushing a new commit with this solution right now.